### PR TITLE
with post call added category

### PR DESCRIPTION
**Pull request by singhsuraj1604**

### CODE REVIEW

**Review by Amittutorials — APPROVED** [review]

File: src/main/java/com/ecom/shopping/adminkart/AdminKartApplication.java [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+import org.apache.ibatis.type.MappedTypes;
+import org.mybatis.spring.annotation.MapperScan;
 import com.ecom.shopping.adminkart.controller.ProductCategoryController;
 import org.apache.ibatis.type.MappedTypes;
 import org.mybatis.spring.annotation.MapperScan;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
-@MappedTypes({ProductCategoryController.class})
+@MappedTypes(value = {ProductCategoryController.class, ProductDto.class})
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/AdminKartApplication.java:16]: try to use only package name apart from DTo class and if working fine then use package name

File: src/main/java/com/ecom/shopping/adminkart/controller/ProductCategoryController.java [hunk 2]
```diff
@@ -14,6 +29,10 @@
 public class ProductCategoryController {
     @Autowired
     private ProductCategoryService productCategoryService;
+    @PostMapping("/addCategory")
+    public void addCategory(@RequestBody ProductCategoryDto productCategoryDto) {
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/controller/ProductCategoryController.java:33]: why return type is void for this method. it should be proper response like ProductCategoryDto

File: src/main/java/com/ecom/shopping/adminkart/controller/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import com.ecom.shopping.adminkart.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/product")
+public class ProductController {
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/controller/ProductController.java:13]: why productcontroller here this PR is only for Category

Comment (smartluckeyboy) [reply src/main/java/com/ecom/shopping/adminkart/controller/ProductController.java:13]: Remove class , not required for this PR

File: src/main/java/com/ecom/shopping/adminkart/dto/ProductDto.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
+package com.ecom.shopping.adminkart.dto;
+
+import java.math.BigDecimal;
+
+public class ProductDto {
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/dto/ProductDto.java:5]: Not required for this PR , plz removed

File: src/main/java/com/ecom/shopping/adminkart/mapper/ProductMapper.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.ecom.shopping.adminkart.mapper;
+
+import com.ecom.shopping.adminkart.dto.ProductDto;
+import org.apache.ibatis.annotations.Param;
+import org.apache.ibatis.annotations.Select;
+import org.apache.ibatis.annotations.Update;
+
+import java.util.Optional;
+
+public interface ProductMapper {
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/mapper/ProductMapper.java:10]: removed for here

File: src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java [hunk 6]
```diff
@@ -5,12 +5,30 @@
...
+import java.util.List;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.List;
 
 @Service
 public class ProductCategoryService {
     @Autowired
-    private ProductCategoryMapper productCategoryMapper;
+    public ProductCategoryMapper productCategoryMapper;
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java:18]: why write two times

File: src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java [hunk 7]
```diff
@@ -5,12 +5,30 @@
...
+    public void addCategory(ProductCategoryDto productCategoryDto) {
+        productCategoryDto.setId(generateId());
+        productCategoryDto.setCreatedDate(getDateTime());
+        productCategoryMapper.insert(productCategoryDto);
+    }
+        private long generateId() {
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java:24]: need to write separate util package

File: src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java [hunk 8]
```diff
@@ -5,12 +5,30 @@
...
+          return(long) Math.floor(Math.random() * 999999999) + 800;
+
+        }
+
+   private String getDateTime() {
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java:29]: same as above comments

File: src/main/java/com/ecom/shopping/adminkart/service/ProductService.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+    @Autowired
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/service/ProductService.java:14]: removed from here

**Review by smartluckeyboy — CHANGES_REQUESTED** [review]
Please check it and incorporate review comments

File: src/main/java/com/ecom/shopping/adminkart/AdminKartApplication.java (hunk 1 above)
Comment (singhsuraj1604) [reply src/main/java/com/ecom/shopping/adminkart/AdminKartApplication.java:16]: without Dto class not working ,please go through below link to check issues.
```
![Screenshot (332)](https://user-images.githubusercontent.com/77046813/226281510-1918b84c-c58a-495c-9f19-39105a6421af.png)

File: src/main/java/com/ecom/shopping/adminkart/controller/ProductController.java (hunk 3 above)
Comment (singhsuraj1604) [reply src/main/java/com/ecom/shopping/adminkart/controller/ProductController.java:13]: please review comment in detail ,didn't get it

**Review by singhsuraj1604 — COMMENTED** [review]
updated

File: src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java (hunk 7 above)
Comment (singhsuraj1604) [reply src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java:23]: ok

**Comment by singhsuraj1604** [comment]
please check changes and merge it

File: src/main/java/com/ecom/shopping/adminkart/mapper/ProductCategoryMapper.java [hunk 10]
```diff
@@ -1,13 +1,18 @@
...
 import com.ecom.shopping.adminkart.dto.ProductCategoryDto;
+import org.apache.ibatis.annotations.Insert;
 import org.apache.ibatis.annotations.Mapper;
 import org.apache.ibatis.annotations.Select;
 
 import java.util.List;
 
 @Mapper
 public interface ProductCategoryMapper {
+    @Insert("insert into product_category(id,name,description,createdDate,modifiedDate,deletedDate) " + "VALUES (#{id},#{name},#{description},#{createdDate},#{modifiedDate},#{deletedDate})")
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/mapper/ProductCategoryMapper.java:12]: remove delete date and modified date

File: src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java [hunk 11]
```diff
@@ -5,12 +5,29 @@
...
     @Autowired
-    private ProductCategoryMapper productCategoryMapper;
+    public ProductCategoryMapper productCategoryMapper;
+    public ProductCategoryDto addCategory(ProductCategoryDto productCategoryDto) {
+        productCategoryDto.setId(generateId());
+        productCategoryDto.setCreatedDate(getDateTime());
+        productCategoryMapper.insert(productCategoryDto);
+        return productCategoryDto;
+    }
+        private long generateId() {
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java:23]: delete this if you have written in util

File: src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java [hunk 12]
```diff
@@ -5,12 +5,29 @@
...
+          return(long) Math.floor(Math.random() * 999999999) + 800;
+
+        }
+
+   private String getDateTime() {
+       return LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS"));
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java:29]: same as above

File: src/main/java/com/ecom/shopping/adminkart/controller/ProductController.java [hunk 13]
```diff
@@ -16,5 +21,8 @@ public class ProductController {
     public ResponseEntity<ProductDto> getProductByCategoryId(@PathVariable long categoryId) {
         ProductDto category=productService.getProductByCategoryId(categoryId);
         return ResponseEntity.ok(category);
+    @PutMapping("/updateProductBy/{id}")
```
Comment (smartluckeyboy) [inline src/main/java/com/ecom/shopping/adminkart/controller/ProductController.java:24]: removed it

**Review by smartluckeyboy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**